Brokerages/BrokerageMultiWebSocketSubscriptionManager.cs

```diff
@@ -0,0 +1,251 @@
+
+                _subscribeFunc(webSocket, symbol, tickType);
+            }
+
+            return true;
+        }
+
+        /// <summary>
+        /// Unsubscribes from the symbols
+        /// </summary>
+        /// <param name="symbols">Symbols to subscribe</param>
+        /// <param name="tickType">Type of tick data</param>
+        protected override bool Unsubscribe(IEnumerable<Symbol> symbols, TickType tickType)
+        {
+            Log.Trace($"BrokerageMultiWebSocketSubscriptionManager.Unsubscribe(): {string.Join(",", symbols.Select(x => x.Value))}");
+
+            foreach (var symbol in symbols)
+            {
+                var entry = GetWebSocketEntryBySymbol(symbol);
+                if (entry != null)
+                {
+                    entry.RemoveSymbol(symbol);
+
+                    _unsubscribeFunc(entry.WebSocket, symbol, tickType);
+                }
+            }
+
+            return true;
+        }
+
+        private BrokerageMultiWebSocketEntry GetWebSocketEntryBySymbol(Symbol symbol)
+        {
+            lock (_locker)
+            {
+                foreach (var entry in _webSocketEntries.Where(entry => entry.Contains(symbol)))
+                {
+                    return entry;
+                }
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        /// Adds a symbol to an existing or new websocket connection
+        /// </summary>
+        private IWebSocket GetWebSocketForSymbol(Symbol symbol)
+        {
+            BrokerageMultiWebSocketEntry entry;
+
+            lock (_locker)
+            {
+                if (_webSocketEntries.All(x => x.SymbolCount >= _maximumSymbolsPerWebSocket))
+                {
+                    if (_maximumWebSocketConnections > 0)
+                    {
+                        throw new NotSupportedException($"Maximum symbol count reached for the current configuration [MaxSymbolsPerWebSocket={_maximumSymbolsPerWebSocket}, MaxWebSocketConnections:{_maximumWebSocketConnections}]");
+                    }
+
+                    // symbol limit reached on all, create new websocket instance
+                    var webSocket = _webSocketFactory();
+                    webSocket.Open += OnOpen;
+
+                    _webSocketEntries.Add(new BrokerageMultiWebSocketEntry(webSocket));
+                }
+
+                // sort by weight ascending, taking into account the symbol limit per websocket
+                _webSocketEntries.Sort((x, y) =>
+                    x.SymbolCount >= _maximumSymbolsPerWebSocket
+                    ? 1
+                    : y.SymbolCount >= _maximumSymbolsPerWebSocket
+                        ? -1
+                        : Math.Sign(x.TotalWeight - y.TotalWeight));
+
+                entry = _webSocketEntries.First();
+            }
+
+            if (!entry.WebSocket.IsOpen)
+            {
+                Connect(entry.WebSocket);
+            }
+
+            entry.AddSymbol(symbol);
```

Comment:
Think `entry` isn't thread safe, since it uses a list internally, `AddSymbol` `RemoveSymbol` should be better wrapped by the lock, or maybe better could make the entry thread safe?